clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -65,6 +65,9 @@
     /* The v2 metadata request is the same as v1. An additional field for cluster id has been added to the v2 metadata response */
     public static final Schema METADATA_REQUEST_V2 = METADATA_REQUEST_V1;
 
+    /* The v3 metadata request is the same as v1 and v2. An additional field for throttle time has been added to the v3 metadata response */
```

Comment:
@ijuma There are a few requests which don't have throttle time (ControlledShutdown, StopReplica etc. used for inter-broker and not by producer/consumer). So it made sense to add throttle time to individual requests. Error codes seem to be handled this way. But my main concern was compatibility. Even if we looked at request version to determine the response header version, there is still the issue of `ApiVersionsResponse` which needs to be handled. According to KIP-35, `ApiVersionResponse.ErrorCode is guaranteed to be the first int16 of the response for all future versions of ApiVersionRequest`. I didn't want to break that assumption.